### PR TITLE
Solution HW 2.2

### PR DESCRIPTION
**Pull request by Bogdan-Demianchuk**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,18 +6,46 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] diagonal = new int[matrix[0].length];
+        for (int i = 0; i < matrix[0].length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: remove empty line

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,18 +6,46 @@
...
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            counterDiagonal[i] = matrix[i][matrix[0].length - i - 1];
+        }
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: same

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -6,18 +6,46 @@
...
-        return 0;
+        int min = matrix[0][0];
+        for (int i = 0; i < matrix.length; i++) {
+            for (int temp : matrix[i]) {
+                if (temp < min) {
+                    min = temp;
+                }
+            }
+        }
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: same

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -6,18 +6,46 @@
...
-        return 0;
+        int max = matrix[0][0];
+        for (int i = 0; i < matrix.length; i++) {
+            for (int temp : matrix[i]) {
+                if (temp > max) {
+                    max = temp;
+                }
+            }
+        }
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: same

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 3 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 4 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**